### PR TITLE
Fix default `shouldCancelWhenOutside` on `Tap` and `LongPress`

### PR DESCRIPTION
**Pull request by Warus15**
## Description

This PR should fix problem mentioned in #2284. `TapGestureHandler` and `LongPressGestureHandler` are now cancelling by default after pointer leaves underlying view, even if `shouldCancelWhenOutside` prop is not passed, as the docs say.

## Test plan

Tested on example app on iOS and Android

### CODE REVIEW

**Comment by j-piasecki** [comment]
IMO doing it in JS is a cleaner approach as it allows to consolidate the default configuration in one place.

For the old API you're able to specify the default config: https://github.com/software-mansion/react-native-gesture-handler/blob/8cd9775e530f342f61d00831f66afe0b3f7b310d/src/handlers/TapGestureHandler.ts#L91 and in the new API you're able to modify relevant config fields in the constructor: https://github.com/software-mansion/react-native-gesture-handler/blob/8cd9775e530f342f61d00831f66afe0b3f7b310d/src/handlers/gestures/tapGesture.ts#L13

I think this way you should also be able to remove this line https://github.com/software-mansion/react-native-gesture-handler/blob/8cd9775e530f342f61d00831f66afe0b3f7b310d/src/web/handlers/TapGestureHandler.ts#L36

The same applies to long press.

**Comment by Warus15** [comment]
You're right @j-piasecki, it is better to put that in default config in JS. I've done this, so please have a look at that when you have time.

File: src/handlers/TapGestureHandler.ts [hunk 1]
```diff
@@ -88,5 +88,7 @@ export const TapGestureHandler = createHandler<
     ...baseGestureHandlerProps,
     ...tapGestureHandlerProps,
   ] as const,
-  config: {},
+  config: {
+    shouldCancelWhenOutside: true,
```
Comment (j-piasecki) [inline src/handlers/TapGestureHandler.ts:92]: This should also be done for long press.

**Review by j-piasecki — COMMENTED** [review]
In`RNGestureHandler.m` there is a lot related to formatting but the imports for Tap and Long press are still there. I didn't look thoroughly at it so there may be more problems. I think undoing the changes there is the simplest solution.

**Comment by Warus15** [comment]
Ok, since this PR got really messed up by this iOS linter changes (which i didn't notice while reverting other commit), I've created new one. When you'll have time, please take a look at #2297, which does what this PR is meant to do, but without any crazy formatting stuff. I'll also close this one.

**Pull request closed without merging**